src/java/org/apache/cassandra/index/sai/utils/BM25Utils.java

```diff
@@ -121,6 +150,8 @@ public static CloseableIterator<PrimaryKeyWithSortKey> computeScores(CloseableIt
                                                                          IndexContext indexContext,
                                                                          Object source)
     {
+        assert source instanceof Memtable || source instanceof SSTableId : "Invalid source " + source.getClass();
+
         // data structures for document stats and frequencies
         ArrayList<DocTF> documents = new ArrayList<>();
```

Comment:
I just noticed that we're assuming sstables are limited to `Integer.MAX_VALUE` many rows, which isn't the case. We might want to consider pushing this logic down to operate on segments instead of on sstables.